code/game/machinery/slotmachine.dm

```diff
@@ -49,10 +49,13 @@
 
 	INVOKE_ASYNC(src, .proc/toggle_reel_spin, FALSE)
 
-	for(cointype in typesof(/obj/item/coin))
-		var/obj/item/coin/C = new cointype
-		coinvalues["[cointype]"] = C.get_item_credit_value()
-		qdel(C) //Sigh
+	if (isnull(coinvalues))
```

Comment:
This could also be done only on demand but I forget how much this costs now (I think like 5ms? Nothing notable)